UefiCpuPkg/PiSmmCpuDxeSmm/X64/PageTbl.c

```diff
@@ -226,10 +225,8 @@ SmmInitPageTable (
 
   //
   // Generate initial SMM page table.
-  // Only map [0, 4G] when PcdCpuSmmRestrictedMemoryAccess is FALSE.
   //
-  PhysicalAddressBits = mCpuSmmRestrictedMemoryAccess ? mPhysicalAddressBits : 32;
-  PageTable           = GenSmmPageTable (mPagingMode, PhysicalAddressBits);
+  PageTable = GenSmmPageTable (mPagingMode, mPhysicalAddressBits);

```

Comment:
Size will be the big problem if always do the full mapping with 1G paging unsupported case.
i'm not sure whether has any platform doesn't support 1g paging...

Reply:
I tested the PR in some Intel SIMICS platform and it worked. Could you specify any client platform that doesn't support 1G page table size?

Reply:
i don't know. i just raise the concern here to make sure we have considered the case. 

if size is not enough, i checked the code, it should hang at the failure of AllocatePageTableMemory in GenSmmPageTable.